guides/source/asset_pipeline.md

```diff
@@ -7,1174 +7,39 @@ This guide covers the asset pipeline.
-and any other environments you define with production behavior (not
-`application.rb`).
-
-TIP: For further details have a look at the docs of your production web server:
-- [Apache](https://tn123.org/mod_xsendfile/)
-- [NGINX](https://www.nginx.com/resources/wiki/start/topics/examples/xsendfile/)
-
-Assets Cache Store
-------------------
-
-By default, Sprockets caches assets in `tmp/cache/assets` in development
-and production environments. This can be changed as follows:
-
-```ruby
-config.assets.configure do |env|
-  env.cache = ActiveSupport::Cache.lookup_store(:memory_store,
-                                                { size: 32.megabytes })
-end
-```
-
-To disable the assets cache store:
-
-```ruby
-config.assets.configure do |env|
-  env.cache = ActiveSupport::Cache.lookup_store(:null_store)
-end
-```
-
-Adding Assets to Your Gems
---------------------------
-
-Assets can also come from external sources in the form of gems.
+### What is different between delivering assets in HTTP1 and HTTP2?
 
-A good example of this is the `jquery-rails` gem.
-This gem contains an engine class which inherits from `Rails::Engine`.
-By doing this, Rails is informed that the directory for this
-gem may contain assets and the `app/assets`, `lib/assets` and
-`vendor/assets` directories of this engine are added to the search path of
-Sprockets.
 
-Making Your Library or Gem a Pre-Processor
-------------------------------------------
+The Four Approaches to the Asset Pipeline
+-----------------------------------------
 
-Sprockets uses Processors, Transformers, Compressors, and Exporters to extend
-Sprockets functionality. Have a look at
-[Extending Sprockets](https://github.com/rails/sprockets/blob/master/guides/extending_sprockets.md)
-to learn more. Here we registered a preprocessor to add a comment to the end
-of text/css (`.css`) files.
+### Sprockets
+The original asset pipeline gem, built for the HTTP/1 era and low javascript frontends. It handled the bundling and digesting of javascript, css and image files, without relying on node packages.
 
-```ruby
-module AddComment
-  def self.call(input)
-    { data: input[:data] + "/* Hello From my sprockets extension */" }
-  end
-end
-```
+### Webpacker
+Shipped with Rails 5.2, as a wrapper around the complexity of Webpack/Node/Yarn, this gem could completely replace Sprockets or simply take over javascript transpiling and bundling. It provided Rails “out of the box” support for SPA frameworks like React.
 
-Now that you have a module that modifies the input data, it's time to register
-it as a preprocessor for your mime type.
+### Import Maps
+Shipped with Rails 7.0, it replaces Sprockets as the default asset pipeline gem. Although it eliminates the need for node/yarn and other complex tooling, it requires the application using it to be deployed in an environment that supports HTTP/2, otherwise it causes severe performance problems.
```

Comment:
It probably should be meantioned that import maps are only for Javascript and not CSS? That's been a big gotcha I've noticed from people asking about it. They are expecting import maps to also handle the CSS.

Reply:
I'd go one step further and tell people what they should use for CSS. Saying that importmaps is just for JS still confuses me when I try to make sense of Rails 7+ assets. The JS story is clear to me, but CSS remains unclear.

Reply:
Just wait until module type assertions come to the browser and then this is no longer true...

🙈 

https://web.dev/css-module-scripts/